### DISCUSSION TITLE
X6如何导出像G6树形list

### DISCUSSION THREAD

**Discussion by zxl0108** [antvis/X6#1342] [Q&A] [answered]
X6如何导出像G6树形list

**Comment by NewByVector** [comment, marked as answer]
可以自己做数据结构转换，x6 默认不支持。

**Reply by qiufeihong2018** [reply]
> X6如何导出像G6树形list

请问有转化的方法提供参考吗？

**Reply by qiufeihong2018** [reply]
我想到了，特来回贴。
举个例子，可以先将X6数据结构转化为普通列表数据结构，然后调用list2Tree方法将列表转化为树结构
```
 /**
   * X6数据结构转化为普通列表数据结构
   * @param graphData X6源数据（有边有节点无pId）
   * @returns
   */
  const graph2List = (graphData: any[]) => {
    graphData.map((item) => {
      if (item.shape === edge) {
        graphData.map((g) => {
          if (g.id === item.target.cell) {
            g.pId = item.source.cell;
          }
        });
      }
    });
    return graphData.filter((g) => g.shape !== edge);
  };
```

**Reply by yuedud** [reply]
> 可以自己做数据结构转换，x6 默认不支持。

有考虑做成数据处理插件吗？
插件里面处理数据，然后toJSON的时候默认都走一遍数据处理函数

**Reply by NewByVector** [reply]
好主意。

**Comment by lloydzhou** [comment]
可以看看[x6-hooks](https://github.com/lloydzhou/x6-hooks)这个包

> 只导出一个useGraphState hooks用于管理x6画布数据
> 同时支持react/vue

在自己的react/vue组件内直接维护一个nodes/edges数组（我相信这两个数组就是你想要的），对x6的操作全都可以转换成setNodes/setEdges操作，这样也就**不需要做导出**。因为你当前组件维护的这个数组本身就是实时更新到x6画布上的。


![image](https://user-images.githubusercontent.com/1826685/219940569-3d4c2149-7fbc-4b96-b277-5c0571e08a09.png)
